tests/ResourceSyncingTest.php

```diff
@@ -763,6 +763,43 @@ function creatingResourceInTenantDatabaseCreatesAndMapInCentralDatabase()
     expect(ResourceUser::first()->role)->toBe('commenter');
 }
 
+test('resources are synced only when sync is enabled', function (bool $enabled) {
+    app()->instance('_tenancy_test_shouldSync', $enabled);
+
+    [$tenant1, $tenant2] = createTenantsAndRunMigrations();
```

Comment:
Is there a specific reason for using two tenants in this test? Instead of just one.

Reply:
I created two tenants when I was testing both sides. Eg creating central and asserting tenant, then creating tenant and asserting central. 

But since you refactored it to the dataset, We can stay with one tenant. You can remove it.

Reply:
I don't think the dataset changes anything here. The code is the same, I just run it with both settings.

But I'm asking about why two tenants are used. Because it seems like the code was using the second tenant just for the `attach()` call, but is there a reason to use tenant2 instead of tenant1 there?

The test in general doesn't test anything about syncing *between tenants*, it just tests syncing between central & tenant contexts. So I'm not sure why two tenants are used.

Reply:
> but is there a reason to use tenant2 instead of tenant1 there?

Yes, there is. 
We are using different tenants for both cases. Because If we use the same tenant, CentralUser would have two records when using the tenant1 for both cases. 
For that, at the bottom of the test, we would have `expect(ResourceUser::all())->toHaveCount(1);` instead of `expect(ResourceUser::all())->toHaveCount(0);`.  It feels more natural to assert count 0 (which means resource not created) instead of  `expect(ResourceUser::all())->toHaveCount(1);`  which means the central user created previously and the count should be 2. Makes sense?

Reply:
I see, yeah that makes sense.